spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/MapTaskExecutionDao.java

```diff
@@ -59,7 +62,12 @@ public void updateTaskExecution(TaskExecution taskExecution) {
 
 	@Override
 	public TaskExecution getTaskExecution(long executionId) {
-		return taskExecutions.get(executionId);
+		if(taskExecutions.containsKey(executionId)) {
```

Comment:
Why check for contains key.  If the map does not have the executionId it should return null.